kythe/web/site/examples.md

```diff
@@ -113,6 +113,90 @@ $ /opt/kythe/tools/runextractor cmake \
     -sourcedir=$CMAKE_ROOT_DIRECTORY
 ```
 
+## Extracting Gradle based repositories
+
+1. Install compiler wrapper
+
+Extraction works by intercepting all calls to `javac` and saving the compiler arguments and inputs to a "compilation unit", which is stored in a .kzip file. We have a javac-wrapper.sh script that forwards javac calls to the java extractor and then calls javac. Add this to the end of your project's build.gradle:
+
+```groovy
+allprojects {
+  gradle.projectsEvaluated {
+    tasks.withType(JavaCompile) {
+      options.fork = true
+      options.forkOptions.executable = '/opt/kythe/extractors/javac-wrapper.sh'
+    }
+  }
+}
+```
+
+2. VName configuration
+
+Next, you will need to create a vnames.json mapping file, which tells the
+extractor how to assign vnames to files based on their paths. A basic vnames
+config for a gradle project looks like:
+
+```json
+// vnames.json
```

Comment:
can we ship a reasonable default file or template for this?

Reply:
I agree we should. I think a good default template would look just like this, but have the java extractor replace the corpus value with something that can be specified via environment variable or flag. We currently do that for bazel, but I don't think the java extractor supports it (yet).

Reply:
It looks like the Java extractor only uses `KYTHE_CORPUS` is `KYTHE_VNAMES` is absent, but should probably use `KYTHE_CORPUS` if the corpus resulting from applying the vnames is empty.

Reply:
I filed this as a separate issue: https://github.com/kythe/kythe/issues/4389.

Reply:
Also #4390.